web/src/topic_list_data.js

```diff
@@ -24,7 +23,7 @@ export function get_list_info(stream_id, zoomed) {
 
     let topic_names = stream_topic_history.get_recent_topic_names(stream_id);
     if (zoomed) {
-        const search_term = topic_list.get_topic_search_term();
+        const search_term = get_topic_search_term();
```

Comment:
Passing a callback is messy -- we should just call the function in the caller and pass the `search_term` string itself.

Reply:
Yeah, that's also a good idea and it will look a lot cleaner as well!